src/lib/common/include/sol-mainloop.h

```diff
@@ -142,6 +142,8 @@ enum sol_fd_flags {
 struct sol_fd;
 struct sol_fd *sol_fd_add(int fd, unsigned int flags, bool (*cb)(void *data, int fd, unsigned int active_flags), const void *data);
 bool sol_fd_del(struct sol_fd *handle);
+bool sol_fd_set_flags(struct sol_fd *handle, unsigned int flags);
```

Comment:
I'd add a helper (static inline) to unset flags, it would do the `set_flags(get_flags() & ~unset_flags`